pkg/session/bootstrap.go

```diff
@@ -1418,35 +1493,41 @@ func checkDistTask(s sessiontypes.Session, ver int64) {
 // upgrade function  will do some upgrade works, when the system is bootstrapped by low version TiDB server
 // For example, add new system variables into mysql.global_variables table.
 func upgrade(s sessiontypes.Session) {
-	ver, err := getBootstrapVersion(s)
+	// Do upgrade works then update bootstrap version.
+	isNull, err := InitMDLVariableForUpgrade(s.GetStore())
+	if err != nil {
+		logutil.BgLogger().Fatal("[upgrade] init metadata lock failed", zap.Error(err))
+	}
+
+	var ver int64
+	i := 0
+	var maxRetryCnt = 10
+	for ; i < maxRetryCnt; i++ {
+		ok := acquireLock(s)
+		if ok {
+			break
+		}
+	}
+	if i == maxRetryCnt {
+		logutil.BgLogger().Fatal("[upgrade] get ddl owner distributed lock failed", zap.Error(err))
+	}
```

Comment:
If acquire lock failed to max retry count，Here should we get version and do upgrade？